it/v2/endpoints/CreateAmendHistoricNonFhlUkPropertyAnnualSubmissionControllerISpec.scala

```diff
@@ -0,0 +1,278 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2.endpoints
+
+import com.github.tomakehurst.wiremock.stubbing.StubMapping
+import play.api.http.HeaderNames.ACCEPT
+import play.api.http.Status._
+import play.api.libs.json.{JsObject, JsValue, Json}
+import play.api.libs.ws.{WSRequest, WSResponse}
+import play.api.test.Helpers.AUTHORIZATION
+import support.V2IntegrationBaseSpec
+import v1.stubs.AuditStub
+import v2.models.errors._
+import v2.stubs.{AuthStub, DownstreamStub, MtdIdLookupStub}
+
+class CreateAmendHistoricNonFhlUkPropertyAnnualSubmissionControllerISpec extends V2IntegrationBaseSpec {
+
+  private trait Test {
+
+    val nino: String          = "AA999999A"
+    val taxYear: String       = "2020-21"
+    val correlationId: String = "X-123"
+
+    val requestBodyJson: JsValue = Json.parse(
+      """
+        |{
+        |   "annualAdjustments": {
+        |      "lossBroughtForward": 200.00,
+        |      "balancingCharge": 200.00,
+        |      "privateUseAdjustment": 200.00,
+        |      "businessPremisesRenovationAllowanceBalancingCharges": 80.02,
+        |      "nonResidentLandlord": true,
+        |      "rentARoom": {
+        |         "jointlyLet": true
+        |      }
+        |   },
+        |   "annualAllowances": {
+        |      "annualInvestmentAllowance": 200.00,
+        |      "zeroEmissionGoodsVehicleAllowance": 200.00,
+        |      "businessPremisesRenovationAllowance": 200.00,
+        |      "otherCapitalAllowance": 200.00,
+        |      "costOfReplacingDomesticGoods": 200.00,
+        |      "propertyIncomeAllowance": 30.02
+        |   }
+        |}
+      """.stripMargin
+    )
+
+    val responseBody: JsValue = Json.parse(
+      s"""
+         |{
+         |   "links":[
+         |      {
+         |         "href":"/individuals/business/property/uk/annual/non-furnished-holiday-lettings/AA999999A/2020-21",
+         |         "method":"GET",
+         |         "rel":"self"
+         |      },
+         |      {
+         |         "href":"/individuals/business/property/uk/annual/non-furnished-holiday-lettings/AA999999A/2020-21",
+         |         "method":"PUT",
+         |         "rel":"create-and-amend-uk-property-historic-non-fhl-annual-submission"
+         |      },
+         |      {
+         |         "href":"/individuals/business/property/uk/annual/non-furnished-holiday-lettings/AA999999A/2020-21",
+         |         "method":"DELETE",
+         |         "rel":"delete-uk-property-historic-non-fhl-annual-submission"
+         |      }
+         |   ]
+         |}
+      """.stripMargin
+    )
+
+    def setupStubs(): StubMapping
+
+    def uri: String = s"/uk/annual/non-furnished-holiday-lettings/$nino/$taxYear"
+
+    def desUri: String = s"/income-tax/nino/$nino/uk-properties/other/annual-summaries/2021"
```

Comment:
can these be `downstreamUri` and `downstreamResponse`? Controller/test shouldn't really "know" about des etc